tests/e2e/network/ensureloadbalancer.go

```diff
@@ -305,6 +305,97 @@ var _ = Describe("Ensure LoadBalancer", func() {
 		Expect(ip).To(Equal(targetIP))
 	})
 
+	// internal w/o IP -> public w/o IP
+	// This test is to replace an upstream k/k one because there's a bug:
+	// https://github.com/kubernetes/kubernetes/blob/373c08e0c7873a76cecde1d6d714cc2ff7af0c9a/test/e2e/network/loadbalancer.go#L574
+	// https://github.com/kubernetes/kubernetes/pull/109413
+	It("should support updating an internal Service to a public one", func() {
+		By("Creating an internal Service")
+		service := utils.CreateLoadBalancerServiceManifest(testServiceName, serviceAnnotationLoadBalancerInternalTrue, labels, ns.Name, ports)
+		_, err := cs.CoreV1().Services(ns.Name).Create(context.TODO(), service, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		utils.Logf("Successfully created LoadBalancer service %s in namespace %s", testServiceName, ns.Name)
+		defer func() {
+			By("Cleaning up")
+			err = utils.DeleteService(cs, ns.Name, testServiceName)
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		By("Waiting for exposure of the internal Service")
+		_, err = utils.WaitServiceExposureAndValidateConnectivity(cs, ns.Name, testServiceName, "")
```

Comment:
I'm wondering if we could check if ip is updated and new connection is sucessful.

Reply:
Updated. PTAL.